.github/workflows/docker.yml

```diff
@@ -0,0 +1,96 @@
+name: Red Hat
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+jobs:
+  centos_stream:
+    name:  ${{ matrix.dist }}-${{ matrix.compiler}}-${{ matrix.build_type }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        dist:
+          - stream9
+          - stream10
+        build_type:
+          - Debug
+          - Release
+        compiler:
+          - gcc
+          - clang
+    container: quay.io/centos/centos:${{ matrix.dist }}
+    steps:
+      - name: Install development tools
+        run: |
+          dnf install -y epel-release
+          /usr/bin/crb enable
+          dnf install -y \
+              clang \
+              gcc \
+              g++ \
+              gfortran \
+              cmake \
+              make \
+              cfitsio-devel \
+              file \
+              git
+
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+
+      - name: Fetch tags from upstream repository
+        run: |
+          git config --global --add safe.directory $(pwd) || true
+          git fetch --tags https://github.com/spacetelescope/hstcal
+
+      - name: build
+        run: |
+          mkdir build
+          cd build
+          cmake .. \
+              -DCMAKE_INSTALL_PREFIX=/tmp/hstcal \
+              -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+              -DCMAKE_C_COMPILER=${{ matrix.compiler }} \
+              -DCMAKE_C_FLAGS="-Wno-incompatible-pointer-types"
+          make -j$(nproc)
+          make install
+
+      - name: Dump file types
+        run: |
+          find /tmp/hstcal -type f | sort | xargs file
+
+      - name: Dump internal version
+        run: |
+          cd build
+          cat << EOF > check_version.c
+          #include <stdio.h>
+          #include "hstcalversion.h"
+          int main(int argc, char *argv[]) {
+              printGitInfo();
+              return 0;
+          }
+          EOF
+          
+          echo "Compiling:"
+          cat -n check_version.c
+          gcc -o check_version -I../include -I. ../lib/hstcalversion.c check_version.c
+          echo
+          echo "Executing:"
+          ./check_version
```

Comment:
Should this just be shipped officially with the package? I feel like generic check version could be useful beyond a smoke test.